engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/JobPreparationActor.scala

```diff
@@ -20,76 +22,83 @@ case class JobPreparationActor(executionData: WorkflowExecutionActorData,
       attributesWithDefault = curriedAddDefaultsToAttributes(evaluatedRuntimeAttributes)
       jobDescriptor = BackendJobDescriptor(workflowDescriptor.backendDescriptor, jobKey, attributesWithDefault, inputs)
-    } yield BackendJobPreparationSucceeded(jobDescriptor, factory.jobExecutionActorProps(jobDescriptor, initializationData, serviceRegistryActor))
+    } yield BackendJobPreparationSucceeded(jobDescriptor, factory.jobExecutionActorProps(jobDescriptor, initializationData, serviceRegistryActor))) match {
+      case Success(s) => s
+      case Failure(f) => BackendJobPreparationFailed(jobKey, f)
+    }
+
   }
 
   // Split inputs map (= evaluated workflow declarations + coerced json inputs) into [init\.*].last
   private lazy val splitInputs = workflowDescriptor.backendDescriptor.inputs map { case (fqn, v) => splitFqn(fqn) -> v }
 
   def resolveAndEvaluateInputs(jobKey: BackendJobDescriptorKey,
-                               wdlFunctions: WdlStandardLibraryFunctions): Try[Map[LocallyQualifiedName, WdlValue]] = {
+                               wdlFunctions: WdlStandardLibraryFunctions): Future[Try[Map[LocallyQualifiedName, WdlValue]]] = {
     import RuntimeAttributeDefinition.buildMapBasedLookup
-    val call = jobKey.call
-    lazy val callInputsFromFile = unqualifiedInputsFromInputFile(call)
-    lazy val workflowScopedLookup = hierarchicalLookup(jobKey.call, jobKey.index) _
-
-    // Try to resolve, evaluate and coerce declarations in order
-    val inputEvaluationAttempt = call.task.declarations.foldLeft(Map.empty[LocallyQualifiedName, Try[WdlValue]])((inputs, declaration) => {
-      val name = declaration.name
-
-      // Try to resolve the declaration, and upon success evaluate the expression
-      // If the declaration is resolved but can't be evaluated this will throw an evaluation exception
-      // If it can't be resolved it's ignored and won't appear in the final input map
-      val evaluated: Option[Try[WdlValue]] = declaration.expression match {
-        // Static expression in the declaration
-        case Some(expr) => Option(expr.evaluate(buildMapBasedLookup(inputs), wdlFunctions))
-        // Expression found in the input mappings
-        case None if call.inputMappings.contains(name) => Option(call.inputMappings(name).evaluate(workflowScopedLookup, wdlFunctions))
-        // Expression found in the input file
-        case None if callInputsFromFile.contains(name) => Option(Success(callInputsFromFile(name)))
-        // Expression can't be found
-        case _ => None
-      }
-
-      // Leave out unresolved declarations
-      evaluated match {
-        case Some(value) =>
-          val coercedValue = value flatMap declaration.wdlType.coerceRawValue
-          inputs + ((name, coercedValue))
-        case None => inputs
-      }
-    })
-
-    TryUtil.sequenceMap(inputEvaluationAttempt, s"Input evaluation for Call ${call.fullyQualifiedName} failed")
+    Future {
```

Comment:
ToL: While the jpa would "see" the abort message sent to receive, I believe the jpa wouldn't/couldn't terminate the runnable block already scheduled via `Future.apply(runnable)(ec)`, as is, since the runnable would be queued up in the ec without a way to kill it. Alternatively, if we _don't_ queue up a runnable on an ec, and use `context stop jpaRef`, then the actor should stop after it finishes the current `receive`, in theory... I haven't fully thought this through and would be interested to hear others thoughts sometime, maybe in a separate ticket.

Blocking the `receive` when an actor has other messages to "simultaneously" process doesn't feel bad yet, but I've been wrong many times before. Everything I've read says that long running tasks should be at bottom of your actor hierarchy, mainly so that if it fails you can restart it, but also with the benefit that the work isn't blocking any important messaging. But- the akka docs seem to hint that if one gets tired of actors, to [just use a Future](http://doc.akka.io/docs/akka/2.4/scala/futures.html#Use_Directly). I'm still looking for best practices with relation to blocking `receive` though.

I'm not sure at the moment what a proper "Abort, stop prepping the job please!" looks like. I did skim these three threads [[1]](https://groups.google.com/d/topic/scala-user/XfT3zsectQI/discussion) [[2]](https://groups.google.com/d/topic/akka-user/TlIbfaC1eb8/discussion) [[3]](https://groups.google.com/d/topic/akka-user/nkD5BN17kVk/discussion), that made it clear that aborting long running akka receives / scala futures should use a hand-coded circuit breaker.